File: frontend/lib/services/steam_service.dart
import '../ffi/protonctl_bridge.dart';
import '../models/steam_candidate.dart';

class SteamService {
  List<SteamCandidate> detectCandidates() =>
      ProtonCtlBridge.detectSteamCandidates().map(SteamCandidate.fromJson).toList();

  SteamCandidate validatePath(String path) =>
      SteamCandidate.fromJson(ProtonCtlBridge.validateSteamPath(path));

  String compatibilityToolsDir(String steamPath) => '$steamPath/compatibilitytools.d';

  String compatDataDir(String steamPath) => '$steamPath/steamapps/compatdata';
}

File: frontend-compose/src/main/kotlin/com/protonctl/app/service/SteamService.kt
package com.protonctl.app.service

import com.protonctl.app.model.SteamCandidate
import com.protonctl.app.native.ProtonCtlBridge

class SteamService {
    fun detectCandidates(): List<SteamCandidate> = ProtonCtlBridge.detectSteamCandidates()

    fun validatePath(path: String): SteamCandidate = ProtonCtlBridge.validateSteamPath(path)

    fun compatibilityToolsDir(steamPath: String): String = "$steamPath/compatibilitytools.d"

    fun compatDataDir(steamPath: String): String = "$steamPath/steamapps/compatdata"
}
